board.h

```diff
@@ -0,0 +1,71 @@
+#include <vector>
+#include <iostream>
+#include <string>
+#include <cstdlib> 
+
+struct Position{
+    int xpos;
+    int ypos;
+
+    bool operator==(const Position b) const{
+        return ((xpos == b.xpos)
+            && (ypos == b.ypos));
+    }
+    Position(){
+
+    }
+    Position(int x, int y){
+        this->xpos = x;
+        this->ypos = y;
+    }
+};
+
+struct Mine : Position{
+    int owner;
+
+    bool operator==(const Mine b) const{

```

Comment:
#should pass `b` as `&` (keeping `const`)

I won't comment about that to avoid spamming, but please review the whole solution and apply the same logic: whenever passing a non-trivial object as parameter, try to do it as a `const&` (or `&` if you need to modify it).